### PR TITLE
Update pydantic to v2 and update datamodel-code-generator to 0.25.6

### PR DESCRIPTION
**Pull request by kumaranvpl**
I saw your PR https://github.com/koxudaxi/fastapi-code-generator/pull/379 and I needed this feature. 
Also I needed pydantic v2 support. 
I forked your branch, installed pydantic v2 and fixed the bugs.
@koxudaxi Please take a look and let me know this is fine or not.

### CODE REVIEW

**Comment by koxudaxi** [comment]
Thanks for the great PR.
To be honest, this project has been a lower priority than my other projects, so I have not been able to devote much resources to it.
I will review it as soon as possible.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -43,9 +44,21 @@ jobs:
...
+    if: github.event.pull_request.draft == false
+    
+    needs:
+      - test
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Decide whether the needed jobs succeeded or failed
+        uses: re-actors/alls-green@release/v1 # nosemgrep
```
Comment (koxudaxi) [inline .github/workflows/test.yml:62]: This is the feature I wanted, and I would like to adopt it in datamodel-code-generator.

File: fastapi_code_generator/parser.py [hunk 2]
```diff
@@ -274,7 +280,10 @@ def get_parameter_type(
         if not data_type:
             if not schema:
                 schema = parameters.schema_
+            if schema is None:
+                raise RuntimeError("schema is None")  # pragma: no cover
             data_type = self.parse_schema(name, schema, [*path, name])
+            data_type = self._collapse_root_model(data_type)
```
Comment (koxudaxi) [inline fastapi_code_generator/parser.py:286]: Obviously, if it were datamodel-code-generator, I would make it optional, but for this project, it should be standard for convenience reasons.

**Review by koxudaxi — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (kumaranvpl) [reply .github/workflows/test.yml:62]: Sure, please do. We use this also in our open source project FastStream - https://github.com/airtai/faststream/blob/main/.github/workflows/test.yaml#L476

**Pull request merged**